solt/transforms/_transforms.py

```diff
@@ -17,13 +17,14 @@
     ALLOWED_COLOR_CONVERSIONS,
     ALLOWED_CROPS,
     ALLOWED_INTERPOLATIONS,
-    ALLOWED_PADDINGS,
     ALLOWED_GRIDMASK_MODES,
 )
 from ..core import Stream
 from ..core import DataContainer, Keypoints
 from ..utils import (
-    img_shape_checker,
+    ensure_valid_image,
```

Comment:
What is the essential difference between an image and the mask? I think that image has always 3 dimensions in the current implementation, and masks have always 2 dimensions

Reply:
Yes, the main difference is that `image` is (d0, d1, ..., dn, ch)`, while `mask` is (d0, d1, ..., dn)`. Still, since these are different types, I believe the check has to be also in different functions. E.g. for mask, it may be reasonable to check whether it is binary, etc.

Reply:
I think it is best if the checks will be related to shapes only. the binary etc checks is the logic of the transform itself. I suggest to make one fucntion `ensure_shape(x, ndim)` that will simply reshape the tensor as needed. I don't think you need to check the incoming shape of the transform at all 🤔